### DISCUSSION TITLE
Support for Display P3 in the online tool

### DISCUSSION THREAD

**Discussion by nesevis** [Myndex/SAPC-APCA#78] [Ideas]
First of all, thank you for the work that has been done on APCA!

This is in reference to the tool at https://www.myndex.com/APCA/

I am in the process of implementing APCA support in [Reveal](https://revealapp.com/)'s accessibility workspace.

Reveal is a tool for iOS developers that lets you inspect your mobile apps, so Display P3 support is very important given its dominance in Apple's devices.

The online tool already supports Display P3 in the sense that the function `displayP3ToY` exists and is available, but there is no way to indicate that you are not using `sRGB` anywhere in the tool's UI.

As part of our implementation, we are linking to the above Myndex URL and passing the colours as query parameters so our users can get more information about APCA and its criteria. 

However, if users are inspecting on their physical device and the colours are given in Display P3, there will be a mismatch between the Lc we display (which is correct and verified) and what the online tool shows because it assumes the values are provided as sRGB.

**Comment by Myndex** [comment]
Hi Chris @nesevis 

Crazy, I wrote this days ago, and apparently never hit the comment button...

-----
Thank you for commenting!

Okay, so the tool, as a demonstrator of the technology, only calculates for sRGB at this time. In part because sRGB is still the default and standard for the web, and the standard for all current accessibility guidelines.

The alternate functions are available in the npm package though, though that leads us into an ambiguous area.

It is not as simple as just "flicking a switch" to change color spaces, but moreover, I am not sure I understand your question?

Are you scraping the APCA tool for results? Or just sending the URL with colors, and sending your users there to view? I ask as the site is not a production site, just a demonstrator.

**Comment by nesevis** [comment]
> It is not as simple as just "flicking a switch" to change color spaces, but moreover, I am not sure I understand your question?

Could there not be a check box that meant that the tool used `displayP3ToY` rather than `sRGBToY` on the input colours before passing the Y values to `APCAcontrast`?

I suppose the displayed colours in the tool would still be in sRGB, which would be confusing in its own way 😅 

> Are you scraping the APCA tool for results? Or just sending the URL with colors, and sending your users there to view? I ask as the site is not a production site, just a demonstrator.

Just sending the URL with query parameters.

**Comment by Myndex** [comment]
Hi Chris @nesevis 

> _I suppose the displayed colours in the tool would still be in sRGB, which would be confusing in its own way 😅_

Exactly. But moreover: as far as web sites are concerned, right now, sRGB is the standard. So even if a user has a P3 display, Apple (at least with Safari) converts web colors to sRGB for display. Only images tagged as P3 are displayed as P3.

This is what I mean when I say it's not just a flick of a switch, and also why that webtool will remain as sRGB fro the time being.

The npm package though has everything needed for doing it natively in-app.
